File: Fortran/ex4/debugInitializeWeights.f90
	subroutine debugInitializeWeights(fan_out, fan_in, W)
	implicit none

	integer:: i, j, k;
	integer::fan_out, fan_in
	double precision, dimension(fan_out, fan_in):: W
	double precision, dimension(fan_out *(fan_in+1)):: T
	
	W = 0

! reshape W

	i = 1
	do j=1, fan_out 
	do k=1, fan_in+1
		T(i)= sin(W(j,k))/10.
		i = i+ 1
	enddo	
	enddo	
	
	i = 1
	do k=1, fan_in+1
	do j=1, fan_out 
		W(j,k) = T(i)
		i = i + 1
	enddo
	enddo		

	end subroutine debugInitializeWeights

File: C/ex4/debugInitializeWeights.c
#include <math.h>

int debugInitializeWeights(int fan_out, int fan_in, double W[fan_out][fan_in])
{

	int i, j, k;
	double T[ fan_out *  fan_in];

	 
	for(i=0; i<fan_out; i++){
		for(j=0; j<fan_in; j++)
			W[i][j] = 0;
	}
  
		k = 0;
	for(i=0; i<fan_out; i++){
		for(j=0; j<fan_in; j++){
			T[k] = sin(W[i][j])/10.;
			k = k + 1;	
		}
	}		

			  k = 0;
	for(j=0; j<fan_in; j++){
		for(i=0; i<fan_out; i++){
			W[i][j] = T[k];	
			k = k + 1;	
		}
	}		

	return 0;
}
